### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by benclewis**

### CODE REVIEW

File: spec/identity_spec.rb [hunk 1]
```diff
@@ -16,24 +19,59 @@
     )
   end
 
-  before do
-    FileUtils.remove_dir(tmp_dir) if File.exists?(tmp_dir)
-    FileUtils.mkdir_p(tmp_dir)
-    File.write(File.join(tmp_dir, 'config'), identity_string)
+  before(:all) do
```
Comment (benclewis) [inline spec/identity_spec.rb:22]: what happened here

Comment (benclewis) [reply spec/identity_spec.rb:22]: the previous version of this worked great so we need to figure out why thats broken now, this is too complex to be maintainable

File: spec/identity_spec.rb [hunk 2]
```diff
@@ -16,24 +19,59 @@
...
+    swarm_port=4061
+    api_port=5061
+    gateway_port=8090
+    @tmp_dir=ENV['TICTAC_TEST_DIR'] || File.join(__dir__, 'tmp')
```
Comment (benclewis) [inline spec/identity_spec.rb:26]: using attributes like this means you do a lot of grepping for the frequently used attributes.
you can get referenceable values like this by adding a let statement like in line 10.
Also your new config appears to be using whatever the user's actual private key is, the tests should use their own.

`let(:cfg) { TicTac::Config.new(tmp_dir)`
`let(:tmp_dir) { ENV['TICTAC_TEST_DIR'] || File.join(__dir__, 'tmp')`

File: spec/identity_spec.rb [hunk 3]
```diff
@@ -16,24 +19,59 @@
...
+    @cfg=TicTac::Config.new(@tmp_dir)
+    puts @tmp_dir
+    config_file=JSON.load(File.open("#{@tmp_dir}/config"))
+    config_file["Addresses"]["Swarm"][0] = "/ip4/127.0.0.1/tcp/#{swarm_port}"
+    config_file["Addresses"]["API"] = "/ip4/127.0.0.1/tcp/#{api_port}"
+    config_file["Addresses"]["Gateway"] = "/ip4/127.0.0.1/tcp/#{gateway_port}"
+    File.write("#{@tmp_dir}/config",JSON.dump(config_file))
+    @external_ipfs=false
```
Comment (benclewis) [inline spec/identity_spec.rb:34]: ditto

File: spec/identity_spec.rb [hunk 4]
```diff
@@ -16,24 +19,59 @@
...
+      ipfs_info=JSON.load(%x(ipfs -c "#{@tmp_dir}" id))
+      if ipfs_info["Addresses"] != nil
+        puts "external client detected"
+        @external_ipfs=true
+      end
+    rescue Exception => e
+      puts "no client detected"
+    end
+    if !@external_ipfs
+      @ipfs_proc=Open3.popen3("ipfs -c #{@tmp_dir} daemon 2>&1")
```
Comment (benclewis) [inline spec/identity_spec.rb:45]: ditto, and maybe this is just me but whenever I see `_proc` in a variable name I assume its callable, so it should respond to #call

Comment (fimbaz) [reply spec/identity_spec.rb:45]: yeah we can call it 'ipfs_process'

File: spec/identity_spec.rb (hunk 1 above)
Comment (fimbaz) [reply spec/identity_spec.rb:22]: Previous version wasn't using IPFS, just checking file paths.  If we wanna assume a running daemon that's fine, but then we shouldn't be making our own config dir.  I've been thinking integration not unit.

**Pull request merged**